app/celery/process_delivery_status_result_tasks.py

```diff
@@ -1,10 +1,135 @@
+import json
+from datetime import datetime
+from celery.exceptions import Retry
 from flask import current_app
-from app import notify_celery
+# import the clients instance from the app
+from notifications_utils.statsd_decorators import statsd
+from app import notify_celery, statsd_client, clients, DATETIME_FORMAT
+from app.config import QueueNames
+from app.dao.notifications_dao import (
+    dao_update_notification,
+    update_notification_status_by_id,
+)
 from app.celery.process_pinpoint_inbound_sms import CeleryEvent
+from app.dao.service_callback import (dao_get_include_status)
+from app.celery.service_callback_tasks import check_and_queue_callback_task
+from app.celery.process_pinpoint_receipt_tasks import attempt_to_get_notification
+import time
 
 
 # Create SQS Queue for Process Deliver Status.
 @notify_celery.task(bind=True, name="process-delivery-status-result", max_retries=48, default_retry_delay=300)
-def process_delivery_status(self, event: CeleryEvent) -> bool:
+@statsd(namespace="tasks")
+def process_delivery_status(self, event: CeleryEvent):
+
+    # log that we are processing the delivery status
     current_app.logger.info('processing delivery status: %s', event)
-    return True
+
+    # first attempt to process the incoming event
+    try:
+        sqs_message = json.loads(event['Message'])
+    except (json.decoder.JSONDecodeError, ValueError, TypeError, KeyError) as e:
+        current_app.logger.exception(e)
+        self.retry(queue=QueueNames.RETRY)
+        return None
+
+    # next parse the information into variables
+    try:
+        # get the provider
+        provider_name = sqs_message.get('provider')
+        provider = clients.get_sms_client(provider_name)
+        body = sqs_message.get('body')
+
+        # get parameters from notification platform status
+        notification_platform_status = provider.translate_delivery_status(body)
+        payload = notification_platform_status.get("payload")
+        reference = notification_platform_status.get("reference")
+        notification_status = notification_platform_status.get("record_status")
+        number_of_message_parts = notification_platform_status.get("number_of_message_parts", 0)
+        price_in_millicents_usd = notification_platform_status.get("price_in_millicents_usd", 0.0)
+
+    except KeyError as e:
+        current_app.logger.error("The event stream message data is missing expected attributes.")
+        current_app.logger.exception(e)
+        current_app.logger.debug(sqs_message)
+        self.retry(queue=QueueNames.RETRY)
+        return None
+
+    current_app.logger.info(
+        "Processing Notification Delivery Status. | reference=%s | notification_status=%s | "
+        "number_of_message_parts=%s | price_in_millicents_usd=%s",
```

Comment:
Should these be `"number_of_message_parts=%d | price_in_millicents_usd=%d"` since one is an int and one a float?